backend/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -70,12 +74,30 @@ private static HashMap<String, Ageable> populateHashMapDB() {
         for (Item item : items) {
             String passedId = String.valueOf(id++);
             switch (item.name) {
-                case "Aged Brie" -> hashMapDB.put(passedId, new AgedBrie(item));
-                case "Backstage passes to a TAFKAL80ETC concert" -> hashMapDB.put(passedId, new BackstagePass(item));
-                case "Sulfuras, Hand of Ragnaros" -> hashMapDB.put(passedId, new Sulfuras(item));
-                case "Conjured" -> hashMapDB.put(passedId, new Conjured(item));
-                case "Red Wine" -> hashMapDB.put(passedId, new RedWine(item));
-                default -> hashMapDB.put(passedId, new DefaultItem(item));
+                case "Aged Brie" -> {
+                    hashMapDB.put(passedId, new AgedBrie(item));
+                    hashMapDB.get(passedId).setId(passedId);
+                }
+                case "Backstage passes to a TAFKAL80ETC concert" -> {
+                    hashMapDB.put(passedId, new BackstagePass(item));
+                    hashMapDB.get(passedId).setId(passedId);
```

Comment:
Question: why do need to do that for all the item type?

Reply:
The way the DataMapper is set up is to return a Collection of Ageable objects. It was simpler to add the ID to each Ageable Item that is created upon initialization. I looked at alternatives, like updating the DataMapper to return the hashmap but that would have a ripple effect across the entire app since the repository returns a List and that List does not include an ID. 

This seemed like the most non-evasive way since this method already generates an ID and passes it as the hashmaps key. Adding it to each object type also allows us to use a setter and getter to dynamically set the ID instead of having to update the constructor to take an ID in, and the getter allows us to grab the ID info for the Json generator.

Unless I am missing a simpler way to do this.

Reply:
I see. Can we have the "database" layer handle setting IDs? For example, instead of directly interacting with hashMap with `hashMap.put`, can we abstract that to `ItemDataSource` which we already instantiated above? For example:
```Java
// ItemsDataSource.java

public class ItemsDataSource<T> implements Datasource<T> {
  ...
  public T insert(Ageable item) {
    // generate id here
    // set ID for the item 
    return hashMapDB.put(id, itemWithId);
  }
}
```
then we can do:
```Java
// GildedRose.java

public static void main(String[] args) {
  ...
  repository = new ItemsDataSource<>(populateHashMapDB());
  itemsController = new ItemsController(repository);
  ...
  ...
  for (Item item : items) {
    String passedId = String.valueOf(id++);
      switch (item.name) {
        case "Aged Brie" -> repository.insert(id, new AgedBrie(item));
        ...
      }
  }
  ...
}
```